encrypt/ipsec_key_rotator.go

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package encrypt
+
+var rotators = map[string]func(key ipsecKey) (ipsecKey, error){
+	"":            func(key ipsecKey) (ipsecKey, error) { return key.rotate() },
+	"gcm-aes":     newGcmAesKey,
+	"hmac-md5":    newHmacMD5Key,
+	"hmac-sha1":   newHmacSHA1Key,
```

Comment:
Are these two options really something we want to encourage using?

Reply:
We just support them and it's up to the user which one should be used.
Do you think it's better to drop them?

Reply:
What do you mean by "we support them"? Technically, we support all algorithms that Linux supports.

I'm mostly unsure we want to encourage MD5 and SHA1 when they are known broken in the general case (although maybe not in this specific authentication case).

Reply:
Well, I put in the map all the algorithms that we used (during testing) for one of our Customers.
I'm happy to raise a PR and remove `hmac-md5` and `hmac-sha1`.